pkg/client/userd/trafficmgr/intercept.go

```diff
@@ -444,20 +563,28 @@ func (tm *TrafficManager) AddIntercept(c context.Context, ir *rpc.CreateIntercep
 	apiPort := uint16(cfg.TelepresenceAPI.Port)
 	if apiPort == 0 {
 		// Default to the API port declared by the traffic-manager
-		apiInfo, err := tm.managerClient.GetTelepresenceAPI(c, &empty.Empty{})
-		if err != nil {
+		var apiInfo *manager.TelepresenceAPIInfo
+		if apiInfo, err = tm.managerClient.GetTelepresenceAPI(c, &empty.Empty{}); err != nil {
 			// Traffic manager is probably outdated. Not fatal, but deserves to be logged
 			dlog.Errorf(c, "failed to obtain Telepresence API info from traffic manager: %v", err)
 		} else {
 			apiPort = uint16(apiInfo.Port)
 		}
 	}
 
-	// It's OK to just call addAgent every time; if the agent is already installed then it's a
-	// no-op.
-	result = tm.addAgent(c, wl, svcprops, ir.AgentImage, apiPort)
-	if result.Error != rpc.InterceptError_UNSPECIFIED {
-		return result, nil
+	var agentEnv map[string]string
+	if svcProps.preparedIntercept == nil {
+		// It's OK to just call addAgent every time; if the agent is already installed then it's a
+		// no-op.
+		agentEnv, result = tm.addAgent(c, svcProps, ir.AgentImage, apiPort)
+		if result.Error != rpc.InterceptError_UNSPECIFIED {
+			return result, nil
+		}
```

Comment:
would we be here bc of an older traffic manager?

Reply:
Yes, the `preparedIntercept` will never be nil when using the new traffic-manager (and always be nil when using the old one). I'll add a comment to clarify.